docs/products/kafka/kafka-connect/howto/clickhouse-sink-connector.md

```diff
@@ -0,0 +1,147 @@
+---
+title: Create a ClickHouse sink connector for Aiven for Apache Kafka®
+sidebar_label: ClickHouse sink connector
+---
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+import ConsoleLabel from "@site/src/components/ConsoleIcons";
+
+The ClickHouse sink connector delivers data from Apache Kafka® topics to a ClickHouse database for efficient querying and analysis.
+
+## Prerequisites
+
+Before you begin, ensure that you have the following:
+
+- An [Aiven for Apache Kafka® service](https://docs.aiven.io/docs/products/kafka/kafka-connect/howto/enable-connect)
+  with Apache Kafka Connect enabled or a [dedicated Aiven for Apache Kafka Connect® service](https://docs.aiven.io/docs/products/kafka/kafka-connect/get-started#apache_kafka_connect_dedicated_cluster).
+- Access to a ClickHouse instance with the following connection details:
+  - Hostname, port, and credentials.
+  - A target database and table that have already been created in ClickHouse.
+
+## Limitations
+
+The ClickHouse sink connector has the following limitations related to data consistency
+and exactly-once delivery:
+
+1. **No exactly-once delivery after restore**: The connector does not guarantee
+   exactly-once delivery after restoring from a backup, powering off, or forking a
+   service. This may result in duplicate records in ClickHouse.
+
+1. **Manual removal of duplicate records**: If duplicate records occur, remove them by
+   running the following command:
+
+   ```sql
+   OPTIMIZE TABLE table_name DEDUPLICATE;
+   ```
+
+    Make sure all potential duplicates are written before running the command to keep
```

Comment:
@harshini-rangaswamy Can you please integrate the following additional steps?

They should be added in order to know that "all potentially duplicates are written":
- Get committed offset on Kafka side:
Kafka instance side menu --> Topics --> Click on the topic used by the connector --> Consumer Group tab --> Offset column

- Get committed offset on Clickhouse side:
Clickhouse side menu --> Query Editor --> Get execute the following query:
`select key, minOffset, maxOffset, state from connect_state;`

Check that:
- `state` column is equal to `AFTER_PROCESSING`
- `minOffset`column is equal to `maxOffset` column
- The offset got from Kafka side is equal to `minOffset`

Reply:
A small correction:

- The offset got from Kafka side is equal **or greater** than minOffset

Reply:
Thanks, @davide-armand. I have included this in the docs.